qualtran/bloqs/block_encoding/product.py

```diff
@@ -132,6 +135,61 @@ def epsilon(self) -> SymbolicFloat:
     def signal_state(self) -> BlackBoxPrepare:
         return BlackBoxPrepare(PrepareIdentity.from_bitsizes([self.ancilla_bitsize]))
 
+    @property
+    def anc_part(self) -> Partition:
+        n = len(self.block_encodings)
+        anc_regs = []
+        if n - 1 > 0:
+            anc_regs.append(Register("flag_bits", dtype=QBit(), shape=(n - 1,)))
+        anc_bits = self.ancilla_bitsize - (n - 1)
+        if is_symbolic(anc_bits) or anc_bits > 0:
+            anc_regs.append(Register("ancilla", dtype=QAny(anc_bits)))
+        return Partition(cast(int, self.ancilla_bitsize), tuple(anc_regs))
+
+    @property
+    def constituents(self) -> Sequence[Bloq]:
+        n = len(self.block_encodings)
+        anc_bits = self.ancilla_bitsize - (n - 1)
+        ret = []
+        for u in reversed(self.block_encodings):
+            partition: List[Tuple[Register, List[Union[str, Unused]]]] = [
+                (Register("system", dtype=QAny(u.system_bitsize)), ["system"])
+            ]
+            if is_symbolic(u.ancilla_bitsize) or u.ancilla_bitsize > 0:
+                regs: List[Union[str, Unused]] = ["ancilla"]
+                if (
+                    is_symbolic(anc_bits)
+                    or is_symbolic(u.ancilla_bitsize)
+                    or anc_bits > u.ancilla_bitsize
+                ):
+                    regs.append(Unused(anc_bits - u.ancilla_bitsize))
+                partition.append((Register("ancilla", dtype=QAny(anc_bits)), regs))
+            if is_symbolic(u.resource_bitsize) or u.resource_bitsize > 0:
+                regs = ["resource"]
+                if is_symbolic(self.resource_bitsize) or self.resource_bitsize > u.resource_bitsize:
+                    regs.append(Unused(self.resource_bitsize - u.resource_bitsize))
+                partition.append((Register("resource", dtype=QAny(u.resource_bitsize)), regs))
+            ret.append(AutoPartition(u, partition, left_only=False))
+        return ret
+
+    def build_call_graph(self, ssa: SympySymbolAllocator) -> Set[BloqCountT]:
+        counts = Counter[Bloq]()
+        if is_symbolic(self.ancilla_bitsize) or self.ancilla_bitsize > 0:
+            counts[self.anc_part] += 1
+            counts[evolve(self.anc_part, partition=False)] += 1
```

Comment:
I think we usually don't count split/join/partition bloqs in `build_call_graph`? @mpharrigan, thoughts?

Reply:
it's more of an art than a science; but I'd say skip it!

Reply:
Removed it.